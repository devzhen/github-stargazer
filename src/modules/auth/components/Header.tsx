import React from 'react';
import styled from 'styled-components/native';

import { IS_IPHONE_X } from '@helpers/deviceInfoHelper';

interface HeaderProps {
  title: string;
  onPress?: () => void;
  bgColor: string;
  withBackNavigation?: boolean;
}

const Header = ({
  title,
  onPress,
  bgColor,
  withBackNavigation,
}: HeaderProps) => (
  <Container bgColor={bgColor}>
    <Title>{title}</Title>
    {withBackNavigation && <BackWrapper onPress={onPress} />}
  </Container>
);

Header.defaultProps = {
  bgColor: '#f7f7f8',
  withBackNavigation: true,
};

const Container = styled.View<{
  bgColor: string;
}>`
  width: 100%;
  height: 64px;
  flex-direction: row;
  justify-content: center;
  align-items: flex-end;
  padding-bottom: 12px;
  background-color: ${(props) => props.bgColor};
  ${IS_IPHONE_X && 'height: 84px'};
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.1);
  z-index: 2;
`;

const Title = styled.Text`
  font-size: 18px;
  font-weight: 500;
`;

const BackWrapper = styled.TouchableOpacity`
  position: absolute;
  left: 16px;
  top: 0;
  height: 64px;
  flex-direction: row;
  align-items: flex-end;
  padding-bottom: 12px;
  ${IS_IPHONE_X && 'top: 20px'};
`;

export default Header;
