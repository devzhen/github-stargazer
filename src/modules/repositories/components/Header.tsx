import React from 'react';
import { Image } from 'react-native';
import styled from 'styled-components/native';

import { IS_IPHONE_X } from '@helpers/deviceInfoHelper';

// @ts-ignore
import backIcon from './assets/arrow_back.png';

interface HeaderProps {
  title: string;
  onPress?: () => void;
  logout?: () => void;
  bgColor: string;
  withBackNavigation?: boolean;
  canLogout?: boolean;
}

const Header = ({
  title,
  onPress,
  bgColor,
  withBackNavigation,
  canLogout,
  logout,
}: HeaderProps) => (
  <Container bgColor={bgColor}>
    <Title numberOfLines={1}>{title}</Title>
    {withBackNavigation && (
      <BackWrapper onPress={onPress}>
        <Image
          source={backIcon}
          style={{ height: 20, width: 20 }}
          resizeMode="contain"
        />
      </BackWrapper>
    )}
    {canLogout && (
      <LogoutWrapper onPress={logout}>
        <Logout>Logout</Logout>
      </LogoutWrapper>
    )}
  </Container>
);

Header.defaultProps = {
  bgColor: '#f7f7f8',
  withBackNavigation: false,
  canLogout: true,
};

const Container = styled.View<{
  bgColor: string;
}>`
  width: 100%;
  height: 64px;
  flex-direction: row;
  justify-content: center;
  align-items: flex-end;
  padding-bottom: 12px;
  background-color: ${(props) => props.bgColor};
  ${IS_IPHONE_X && 'height: 84px'};
  border-bottom-width: 1px;
  border-bottom-color: ${(props) => props.theme.colors.gray};
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.1);
  z-index: 2;
`;

const Title = styled.Text`
  font-size: 18px;
  font-weight: 500;
`;

const BackWrapper = styled.TouchableOpacity`
  position: absolute;
  left: 16px;
  top: 0;
  height: 64px;
  flex-direction: row;
  align-items: flex-end;
  padding-bottom: 12px;
  ${IS_IPHONE_X && 'top: 20px'};
`;

const LogoutWrapper = styled.TouchableOpacity`
  position: absolute;
  right: 0px;
  top: 0;
  height: 64px;
  flex-direction: row;
  align-items: center;
  padding-right: 16px;
`;

const Logout = styled.Text`
  font-size: 18px;
  font-weight: 500;
  color: ${(props) => props.theme.colors.orange};
`;

export default Header;
