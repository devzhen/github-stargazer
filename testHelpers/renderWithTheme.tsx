import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import 'jest-styled-components';

import theme from '@theme/index';

const renderWithTheme = (component: any) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export default renderWithTheme;
