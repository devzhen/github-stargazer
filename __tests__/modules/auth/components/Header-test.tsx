import React from 'react';
import 'jest-styled-components';

import Header from '@modules/auth/components/Header';

// @ts-ignore
import renderWithTheme from '@testHelpers/renderWithTheme';

describe('Header auth component', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Header title="Login" bgColor="#fff" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
