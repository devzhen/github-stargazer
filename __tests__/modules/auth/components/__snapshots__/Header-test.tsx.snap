// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header auth component renders correctly 1`] = `
<View
  bgColor="#fff"
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "center",
        "paddingBottom": 12,
        "shadowColor": "rgba(0, 0, 0, 0.1)",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": "100%",
        "zIndex": 2,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "fontWeight": "500",
        },
      ]
    }
  >
    Login
  </Text>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "height": 64,
        "left": 16,
        "opacity": 1,
        "paddingBottom": 12,
        "position": "absolute",
        "top": 0,
      }
    }
  />
</View>
`;
